File: src/lib/types/stations.ts
export type StationType = "sub2api" | "newapi" | "openai-compatible" | "custom";

export type StationStatus =
  | "healthy"
  | "warning"
  | "error"
  | "disabled"
  | "unchecked";

export type StationProxyMode = "inherit" | "direct" | "system" | "manual";

export type Station = {
  id: string;
  name: string;
  stationType: StationType;
  websiteUrl: string;
  apiBaseUrl: string;
  endpointRevision: number;
  collectorProxyMode: StationProxyMode;
  collectorProxyUrl: string | null;
  apiKeyMasked: string;
  apiKeyPresent: boolean;
  keyCount: number;
  enabled: boolean;
  priority: number;
  creditPerCny: number;
  balanceRaw: number | null;
  balanceCny: number | null;
  lowBalanceThresholdCny: number | null;
  collectionIntervalMinutes: number;
  status: StationStatus;
  latencyMs: number | null;
  lastCheckedAt: string | null;
  lastPricingFetchedAt: string | null;
  note: string | null;
  createdAt: string;
  updatedAt: string;
};

export type StationInput = {
  name: string;
  stationType: StationType;
  websiteUrl: string;
  apiBaseUrl: string;
  apiKey: string;
  collectorProxyMode: StationProxyMode;
  collectorProxyUrl: string | null;
  enabled: boolean;
  creditPerCny: number;
  lowBalanceThresholdCny: number | null;
  collectionIntervalMinutes: number;
  note: string | null;
};

export type StationUpdateInput = Omit<StationInput, "apiKey"> & {
  id: string;
  apiKey: string | null;
};

export type StationEndpointHealth = {
  stationId: string;
  status: "unchecked" | "success" | "failed";
  latencyMs: number | null;
  checkedAt: string | null;
  errorSummary: string | null;
  updatedAt: string;
};

export type EndpointPingResult = {
  stationId: string;
  ok: boolean;
  status: "success" | "failed";
  latencyMs: number | null;
  checkedAt: string;
  errorSummary: string | null;
};

export const stationTypeLabels: Record<StationType, string> = {
  sub2api: "Sub2API",
  newapi: "NewAPI",
  "openai-compatible": "自定义接口",
  custom: "自定义接口",
};

export const stationTypeOptions: Array<{ value: StationType; label: string }> = [
  { value: "sub2api", label: stationTypeLabels.sub2api },
  { value: "newapi", label: stationTypeLabels.newapi },
  { value: "custom", label: stationTypeLabels.custom },
];

export const stationStatusLabels: Record<StationStatus, string> = {
  healthy: "采集正常",
  warning: "采集需关注",
  error: "采集异常",
  disabled: "禁用",
  unchecked: "未采集",
};

export const stationProxyModeLabels: Record<StationProxyMode, string> = {
  inherit: "继承全局",
  direct: "直连",
  system: "使用系统代理",
  manual: "手动代理地址",
};

File: src-tauri/src/models/stations.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct Station {
    pub id: String,
    pub name: String,
    pub station_type: String,
    pub website_url: String,
    pub api_base_url: String,
    pub endpoint_revision: i64,
    pub collector_proxy_mode: String,
    pub collector_proxy_url: Option<String>,
    pub api_key_masked: String,
    pub api_key_present: bool,
    pub key_count: i64,
    pub enabled: bool,
    pub priority: i64,
    pub credit_per_cny: f64,
    pub balance_raw: Option<f64>,
    pub balance_cny: Option<f64>,
    pub low_balance_threshold_cny: Option<f64>,
    pub collection_interval_minutes: u16,
    pub status: String,
    pub latency_ms: Option<i64>,
    pub last_checked_at: Option<String>,
    pub last_pricing_fetched_at: Option<String>,
    pub note: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct StationEndpointHealth {
    pub station_id: String,
    pub endpoint_revision: i64,
    pub status: String,
    pub latency_ms: Option<i64>,
    pub checked_at: Option<String>,
    pub error_summary: Option<String>,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct EndpointPingResult {
    pub station_id: String,
    pub ok: bool,
    pub status: String,
    pub latency_ms: Option<i64>,
    pub checked_at: String,
    pub error_summary: Option<String>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateStationInput {
    pub name: String,
    pub station_type: String,
    pub website_url: String,
    pub api_base_url: String,
    pub api_key: String,
    pub collector_proxy_mode: String,
    pub collector_proxy_url: Option<String>,
    pub enabled: bool,
    pub credit_per_cny: f64,
    pub low_balance_threshold_cny: Option<f64>,
    pub collection_interval_minutes: u16,
    pub note: Option<String>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateStationInput {
    pub id: String,
    pub name: String,
    pub station_type: String,
    pub website_url: String,
    pub api_base_url: String,
    pub api_key: Option<String>,
    pub collector_proxy_mode: String,
    pub collector_proxy_url: Option<String>,
    pub enabled: bool,
    pub credit_per_cny: f64,
    pub low_balance_threshold_cny: Option<f64>,
    pub collection_interval_minutes: u16,
    pub note: Option<String>,
}
